kevm-pyk/src/kevm_pyk/foundry.py

```diff
@@ -375,12 +360,12 @@ def update_kompilation_digest() -> None:
 
     if not kompilation_up_to_date() or rekompile or not kompiled_timestamp.exists():
         _LOGGER.info(f'Kompiling definition: {foundry_main_file}')
-        _kompile(foundry_definition_dir, KompileBackend.HASKELL, md_selector=md_selector)
+        _kompile(foundry_definition_dir, KompileTarget.HASKELL, md_selector=md_selector)
         if llvm_library:
             _LOGGER.info(f'Kompiling definition to LLVM dy.lib: {foundry_main_file}')
             _kompile(
                 foundry_llvm_dir,
-                KompileBackend.LLVM,
+                KompileTarget.LLVM,
                 llvm_kompile_type=LLVMKompileType.C,
                 md_selector=('k & ! symbolic' if md_selector is None else f'{md_selector} & ! symbolic'),
```

Comment:
@ehildenb we pass custom md_selector here

Reply:
I think we should instead modify the `class KompileTarget` to have a new property (in file `kompile.py`):

```
@property
def md_selector(self) -> str:
        match self:
            case self.LLVM:
                return 'k & ! node & ! symbolic'
            case self.NODE:
                return 'k & ! symbolic & ! standalone'
            case self.HASKELL:
                return 'k & ! node & ! concrete'
            case self.HASKELL_STANDALONE:
                return 'k & ! node & ! concrete'
            case self.FOUNDRY:
                return 'k & ! node & ! concrete'
            case _:
                raise AssertionError()
```

Then have function `kevm_kompile` (also defined in `kompile.py`) just use this method. That way we don't have to pass in custom `md_selector`, since it's fully a configuration value dependent on the compilation target, not something the user should really be messing with directly for the moment.

Reply:
so just to check, we want to completely remove the md_selector as a flag that the user can specify as a command line arg and instead just pass the right flags based on the compile target?

Reply:
That seems sensible to me; arbitrary changes to the `md_selector` that aren't sanctioned by the design of the semantics are likely to just break things, I think.